enhancements/network/configure-ovs-alternative.md

```diff
@@ -0,0 +1,568 @@
+        mode: 0644
+        overwrite: true
+        path: /etc/nmstate/cluster.yaml
+```
+
+The only difference is there will be a single file applied to each node of the
+specified role. The name of the file must be `cluster.yaml`.
+If there are both node-specific and cluster-wide configuration
+files present, the node-specific one will take precedence. Again, separate
+manifests will be needed for each role because of how MCO works.
+
+When this mechanism is in use, a flag will be set to indicate to configure-ovs
+that it should not attempt to configure br-ex.
+
+If changes to the bridge configuration are desired on day 2 (e.g. adding a VLAN
+or changing MTU), the user will need to install Kubernetes-NMState and use it
+to apply those changes directly to br-ex. The configuration applied initially
+should be copied into an NNCP and used as the basis for the update config.
+Modifying the machine-config for an already-deployed node will have no effect.
+
+#### Scaleout
+
+When scaling out new nodes on day 2, it will be necessary to update the
+appropriate machine-configs with NMState YAML for the new node(s). If a
+cluster-wide configuration is in use, no changes will be necessary. This
+must happen before the scaleout occurs.
+
+Since we don't want to trigger a reboot of the entire cluster before each
+scaleout operation, we will add the NMState configuration directory to the
+list of paths for which the machine-config-operator will not trigger a reboot.
+Files in this path will only be applied on initial deployment anyway so it
+will never be necessary to reboot a node when updates are made. Note that,
+as mentioned above, if changes are desired on day 2 they should be made
+with Kubernetes-NMState, not by updating the machine-configs used here.
+
+#### Variation [optional]
+
+We have two paths forward for applying the configuration. One is to write a
+custom service that will select only the host-specific NMState file and
+apply it. This is what my [current prototype](https://docs.google.com/document/d/1Zp1J2HbVvu-v4mHc9M594JLqt6UwdKoix8xkeSzVL98/edit#heading=h.fba4j9nvp0nl) does.
+
+The other is to use the NMState service already included in the OS image to
+apply the configurations. One issue with this is that it will apply everything
+in the /etc/nmstate directory rather than just the host-specific file.
+To get around that, we could still implement a custom service, but instead of
+applying the configurations, it would only be responsible for copying the
+correct file to /etc/nmstate so the regular service can apply it. This has
+the benefit of not reinventing the NMState service and still gives us full
+control over what gets applied to the host, and also leaves us a way to signal
+to ovs-configuration that we've already handled the bridge setup.
```

Comment:
We generally don't support mixing network configuration mechanisms. Still, it would probably be safest to verify there is no conflict with files already in /etc/nmstate to avoid overwriting a file we shouldn't. In that case we should probably just fail the service.